zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/migration/DbMigratorImpl.java

```diff
@@ -77,49 +79,70 @@ public DbMigratorImpl(
     this(
         versionCheckRestrictionEnabled,
         new MigrationTaskContextImpl(clusterContext, processingState),
-        MIGRATION_TASKS);
+        MIGRATION_TASKS,
+        null);
```

Comment:
🔧 Currently the `VisibleForTesting` constructor doesn't specify what the behavior is when currentVersion is `null`, so it requires a bit of back-and-forth reading to understand it. You could consider passing `VersionUtil.getVersion()` here. Will also require the change on line 89, so there's some duplication. Benefit is that it's always clear to the reader what this value is.

Reply:
Yes, that was actually my original implementation. But in the end I think it's better to have the default in just one place (the main constructor), as you suggest as well.
I think for the reader it's quite quick to jump to the main constructor and see what's the default there :shrug: